### PR TITLE
integrate aie-pdi-transform into xclbinutil

### PR DESCRIPTION
**Pull request by xfreid**
#### Problem solved by the commit
Integrate aie-pdi-transform to xclbinutil per Larry's request

The request was from Larry Liu and Sonal Santan
"All current shipping xclbins need to be updated to use the new format.
We need to integrate [aie-pdi-transform] (https://gitenterprise.xilinx.com/ai-engine/aie-pdi-transform) into xclbinutil so we can upgrade an existing xclbin into the new format using xclbinutil. The details of the tools are here—
https://confluence.xilinx.com/pages/viewpage.action?spaceKey=~huaj&title=Transform+PDI+In+IPU-PHX"

The code is from aie-pdi-transform git repo - https://gitenterprise.xilinx.com/ai-engine/aie-pdi-transform
This code was initially developed and maintained by Hua, but I think he is no longer responsible for maintaining this repo. 

Per Sonal's guidance, we copied the source code from the repo above to XRT so that transform_static can be built and shipped along with xclbinutil

This repo is only used for IPU, once we copy the code to XRT repo under xclbinutil, the source code in XRT becomes the single source of truth, so there won’t be an out-of-sync concern.

The source code in aie-pdi-transform is going through legal sweep right now. [@Varughese, Irene](mailto:Irene.Varughese@amd.com) will help us to get legal approval once all the flagged items are addressed.

The details about “what does it do, why, and how” can be found in https://confluence.xilinx.com/pages/viewpage.action?spaceKey=~huaj&title=Transform+PDI+In+IPU-PHX

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
N/A

#### How problem was solved, alternative solutions (if any) and why they were rejected
N/A

#### Risks (if any) associated the changes in the commit
N/A

#### What has been tested and how, request additional testing if necessary
Manual test
Ran clang-tidy to fix the reported warning

#### Documentation impact (if any)
None

### CODE REVIEW

**Comment by xfreid** [comment]
Hi @sonals , the previous PR https://github.com/Xilinx/XRT/pull/7993 was closed by Soren. Phil asked me to run clang-tidy on this newly added sub-folder "aie-pdi-transform" and fix all the generated warnings.
I used the `.clang-tidy` at the root directory as the template and removed the checkers related to C++ (as shown below)
`cppcoreguidelines-*,
hicpp-*,
-hicpp-no-array-decay,
-hicpp-signed-bitwise,
-cppcoreguidelines-pro-bounds-pointer-arithmetic,
-cppcoreguidelines-pro-type-union-access,
-cppcoreguidelines-pro-bounds-array-to-pointer-decay,
-cppcoreguidelines-pro-type-reinterpret-cast,
-cppcoreguidelines-pro-bounds-constant-array-index,`
I fixed all the warning messages, please review

The source files are still going through the legal scan. If everything looks good, please approve, but hold off Merge until we get the legal approval. Thanks

**Review by sonals — CHANGES_REQUESTED** [review]
xclbinutil should link with the aie-pdi-transform objects files. Having another executable to perform the transform complicates things.

**Comment by xfreid** [comment]
The release request has been approved by legal ([SCL-2409](https://jira.xilinx.com/browse/SCL-2409)).

File: src/runtime_src/tools/xclbinutil/aie-pdi-transform/.clang-tidy
Comment (sonals) [inline src/runtime_src/tools/xclbinutil/aie-pdi-transform/.clang-tidy:1]: Do we need this separate copy of .clang-tidy? I believe clang-tidy recursively searches for .clang-tidy in the parent directories so it should find the one in the root of the project.

File: src/runtime_src/tools/xclbinutil/XclBinUtilities.cxx
Comment (sonals) [inline src/runtime_src/tools/xclbinutil/XclBinUtilities.cxx:1019]: A lot of hairy low level C code ;-)

File: src/runtime_src/tools/xclbinutil/XclBinUtilities.cxx
Comment (sonals) [inline src/runtime_src/tools/xclbinutil/XclBinUtilities.cxx:1028]: We are using the same filename for both input and output?

File: src/runtime_src/tools/xclbinutil/aie-pdi-transform/libinclude/cdo_common.h
Comment (sonals) [inline src/runtime_src/tools/xclbinutil/aie-pdi-transform/libinclude/cdo_common.h:29]: Can we change this from printf to fprintf. That will give us an option to capture all the output in a log file. We can specify the log file name as third arg in  pdi_transform(). pdi_transform() should open the log file a setup the global file pointer used in XCdo_Print()?

File: src/runtime_src/tools/xclbinutil/XclBinUtilities.cxx
Comment (xfreid) [reply src/runtime_src/tools/xclbinutil/XclBinUtilities.cxx:1028]: FILE* file here is only used to redirect the stdout so that output of pdi_transform() won't pollute the console
It has nothing to do with input

File: src/runtime_src/tools/xclbinutil/aie-pdi-transform/.clang-tidy
Comment (xfreid) [reply src/runtime_src/tools/xclbinutil/aie-pdi-transform/.clang-tidy:1]: This .clang-tidy is different from the one in the root directory. I added a comment a while ago.
> I used the .clang-tidy at the root directory as the template and removed the checkers related to C++ (as shown below)
> cppcoreguidelines-*, hicpp-*, -hicpp-no-array-decay, -hicpp-signed-bitwise, -cppcoreguidelines-pro-bounds-pointer-arithmetic, -cppcoreguidelines-pro-type-union-access, -cppcoreguidelines-pro-bounds-array-to-pointer-decay, -cppcoreguidelines-pro-type-reinterpret-cast, -cppcoreguidelines-pro-bounds-constant-array-index,

File: src/runtime_src/tools/xclbinutil/aie-pdi-transform/libinclude/cdo_common.h
Comment (xfreid) [reply src/runtime_src/tools/xclbinutil/aie-pdi-transform/libinclude/cdo_common.h:29]: I can try to change printf to fprintf

**Pull request merged**
